### DISCUSSION TITLE
Why are BelongsTo relationships based off the method name, and not the model name?

### DISCUSSION THREAD

**Discussion by SebastianSlangerup** [laravel/framework#52728] [Q&A]
The documentation reads:
> Eloquent determines the foreign key name by examining the name of the relationship method and suffixing the method name with _id. So, in this case, Eloquent assumes that the Phone model has a user_id column.

But I was curious and I'd like to know more about the technical side of this, if there are any...
Why the method name? Wouldn't it make more sense to base this relationship off of the model passed into the method instead?

**Comment by ghost** [comment]
You are perfectly right. I never let laravel guess the parameters of the relation definition and I did not saw this before. Good catch.

**Reply by ghost** [reply]
It should base it on the table name not on the class name.

**Comment by kokoshneta** [comment]
I asked [the same question](https://github.com/laravel/framework/discussions/49569) about a year ago and got no more feedback than you have here.

This remains one of the bizarrest design choices in Laravel to me. Completely unintuitive, breaks consistency, has zero practical benefits (as far as I can see)… just all-around weird.

**Reply by ghost** [reply]
Like I said. ALWAYS write all params for relations.  It will make the code even faster. (This will never be fixed for backward compatibility reasons.)
